include/zephyr/net/mdio.h

```diff
@@ -0,0 +1,233 @@
+
+/* MDIO generic registers */
+/** Control 1 */
+#define MDIO_CTRL1			0
+/** Status 1 */
+#define MDIO_STAT1			1
+/** Device identifier (1) */
+#define MDIO_DEVID1			2
+/** Device identifier (2) */
+#define MDIO_DEVID2			3
+/** Speed ability */
+#define MDIO_SPEED			4
+/** Devices in package (1) */
+#define MDIO_DEVS1			5
+/** Devices in package (2) */
+#define MDIO_DEVS2			6
+/** Control 2 */
+#define MDIO_CTRL2			7
+/** Status 2 */
+#define MDIO_STAT2			8
+/** Package identifier (1) */
+#define MDIO_PKGID1			14
+/** Package identifier (2) */
+#define MDIO_PKGID2			15
+
+
+/* BASE-T1 registers */
+/** BASE-T1 Auto-negotiation control */
+#define MDIO_AN_T1_CTRL			512
+/** BASE-T1 Auto-negotiation status */
+#define MDIO_AN_T1_STAT			513
+/** BASE-T1 Auto-negotiation advertisement register [15:0] */
+#define MDIO_AN_T1_ADV_L		514
+/** BASE-T1 Auto-negotiation advertisement register [31:16] */
+#define MDIO_AN_T1_ADV_M		515
+/** BASE-T1 Auto-negotiation advertisement register [47:32] */
+#define MDIO_AN_T1_ADV_H		516
+
+/* BASE-T1 Auto-negotiation Control register */
+/** Auto-negotiation Restart */
+#define MDIO_AN_T1_CTRL_RESTART		BIT(9)
+/** Auto-negotiation Enable */
+#define MDIO_AN_T1_CTRL_EN		BIT(12)
+
+/* BASE-T1 Auto-negotiation Status register */
+/** Link Status */
+#define MDIO_AN_T1_STAT_LINK_STATUS	BIT(2)
+/** Auto-negotiation Ability */
+#define MDIO_AN_T1_STAT_ABLE		BIT(3)
+/** Auto-negotiation Remote Fault */
+#define MDIO_AN_T1_STAT_REMOTE_FAULT	BIT(4)
+/** Auto-negotiation Complete */
+#define MDIO_AN_T1_STAT_COMPLETE	BIT(5)
+/** Page Received */
+#define MDIO_AN_T1_STAT_PAGE_RX		BIT(6)
+
+/* BASE-T1 Auto-negotiation Advertisement register [15:0] */
+/** Pause Ability */
+#define MDIO_AN_T1_ADV_L_PAUSE_CAP	BIT(10)
+/** Pause Ability */
+#define MDIO_AN_T1_ADV_L_PAUSE_ASYM	BIT(11)
+/** Force Master/Slave Configuration */
+#define MDIO_AN_T1_ADV_L_FORCE_MS	BIT(12)
+/** Remote Fault */
+#define MDIO_AN_T1_ADV_L_REMOTE_FAULT	BIT(13)
+/** Acknowledge (ACK) */
+#define MDIO_AN_T1_ADV_L_ACK		BIT(14)
+/** Next Page Request */
+#define MDIO_AN_T1_ADV_L_NEXT_PAGE_REQ	BIT(15)
+
+/* BASE-T1 Auto-negotiation Advertisement register [31:16] */
+/** 10BASE-T1L Ability */
+#define MDIO_AN_T1_ADV_M_B10L		BIT(14)
+/** Master/slave Configuration */
+#define MDIO_AN_T1_ADV_M_MST		BIT(4)
+
+/* BASE-T1 Auto-negotiation Advertisement register [47:32] */
+/* 10BASE-T1L High Level Transmit Operating Mode Request */
+#define MDIO_AN_T1_ADV_H_10L_TX_HI_REQ	BIT(12)
+/* 10BASE-T1L High Level Transmit Operating Mode Ability */
+#define MDIO_AN_T1_ADV_H_10L_TX_HI	BIT(13)
+
+
+/* 10BASE-T1L registers */
+/** 10BASE-T1L PMA control */
+#define MDIO_PMA_B10L_CTRL		2294
+/** 10BASE-T1L PMA status */
+#define MDIO_PMA_B10L_STAT		2295
+/** 10BASE-T1L PMA link status*/
+#define MDIO_PMA_B10L_LINK_STAT		33538
+/** 10BASE-T1L PCS control */
+#define MDIO_PCS_B10L_CTRL		2278
+/** 10BASE-T1L PCS status */
+#define MDIO_PCS_B10L_STAT		2278
```

Comment:
Please let's keep using hex for register addresses.
Values of  `MDIO_PCS_B10L_CTRL` and `MDIO_PCS_B10L_STAT` are different.

```suggestion
#define MDIO_PMA_B10L_CTRL		0x08F6U
/** 10BASE-T1L PMA status */
#define MDIO_PMA_B10L_STAT		0x08F7U
/** 10BASE-T1L PMA link status*/
#define MDIO_PMA_B10L_LINK_STAT		0x8302U
/** 10BASE-T1L PCS control */
#define MDIO_PCS_B10L_CTRL		0x08E6U
/** 10BASE-T1L PCS status */
#define MDIO_PCS_B10L_STAT		0x08E7U
```

Reply:
> Please let's keep using hex for register addresses.

I've used decimal values as this is the representation used in the standard and as a reference also in Linux headers (https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/mdio.h#L70).

> Values of MDIO_PCS_B10L_CTRL and MDIO_PCS_B10L_STAT are different.
Hmm I fail to see how the values you are proposing are different than the ones in the code, beside the hex and decimal representation? e.g. 0x08E6 = 2278

Reply:
ah ok I got where's the error, `MDIO_PCS_B10L_STAT` should be `2279` (0x08E7U)